test/test_client.py

```diff
@@ -239,6 +266,60 @@ def test_should_return_null_if_transport_fails(self, dummy_thread_class):
         expect(self._subject.check(dummy_request)).to(be_none)
 
 
+class TestClientQuota(unittest2.TestCase):
+    SERVICE_NAME = u'quota'
+    PROJECT_ID = SERVICE_NAME + u'.project'
+
+    def setUp(self):
+        self._mock_transport = mock.MagicMock()
+        self._subject = client.Loaders.DEFAULT.load(
+            self.SERVICE_NAME,
+            create_transport=lambda: self._mock_transport)
+
+    def test_should_raise_on_quota_without_start(self):
+        dummy_request = _make_dummy_quota_request(self.PROJECT_ID,
+                                                  self.SERVICE_NAME)
+        expect(lambda: self._subject.allocate_quota(dummy_request)).to(
+            raise_error(AssertionError))
+
+    @mock.patch(u"endpoints_management.control.client._THREAD_CLASS", spec=True)
+    def test_should_queue_the_request_if_not_cached(self, dummy_thread_class):
```

Comment:
Any reason to set `spec=True`, but not use `dummy_thread_class` at all?

Reply:
This class started as a copy of TestClientCheck, so I have to speculate. Patching out these classes does have a benefit even if you don't do anything with the mock; it means calls to start new threads won't actually start new threads.

Reply:
Okay. I guess I'm asking if it would be any different with just the following?
```
@mock.patch(u"endpoints_management.control.client._THREAD_CLASS")
def test_should_queue_the_request_if_not_cached(self):
  ...
```

Reply:
Yeah, I'm pretty sure we'd get errors because of the missing argument to the method.